### ISSUE TITLE
Maybe it's better to add Utf8PathBuf::as_std_pathbuf, and Utf8PathBuf::as_std_path?

### ISSUE THREAD

**Issue by WindSoilder** [withoutboats/camino#8]
Refer to #3 , I found these api is helpful when we want to convert from Utf8Path to Path(because some api needs it), and maybe it's also useful to convert from Utf8PathBuf to PathBuf?

If it's okey, I can try to help on this :-)

**Comment by WindSoilder** [comment]
Oh, I've found that #5 revert back `as_std_path` impl, and use `as_ref()` instead, sorry for disturb, close the issue for now :-)

**Comment by sunshowers** [comment]
Thanks! Yeah AsRef seems like the logical way to do this for now.

**Comment by Emerentius** [comment]
I converted a small project to use camino and it was pretty painless for the most part, but the biggest friction came exactly from the lack of non-generic methods to convert between `std` paths and utf8 paths.

Any instance of this pattern meant I needed to introduce a new temporary variable: `utf8path.as_ref().some_method_on_path()` or `path.try_into()?.some_method_on_utf8path()`. That was a bit annoying.
The need for the conversions came because 1) `read_dir()` still returns std paths and 2) external crates work with std paths, namely `fs_err` and `walkdir` in my case.
